Getting linker errors with QOS using self-built fastrtps in Visual Studio project [6216]

I downloaded the source for fastrtps and it's dependencies to compile my own build of 1.8.1. I followed the instructions in the README and successfully built, but when trying to use fastrtps in my code, I get linker errors in visual studio.

```
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::DurabilityQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@DurabilityQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::DeadlineQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@DeadlineQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::LatencyBudgetQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@LatencyBudgetQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::LivelinessQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@LivelinessQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::ReliabilityQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@ReliabilityQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::OwnershipQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@OwnershipQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::DestinationOrderQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@DestinationOrderQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::UserDataQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@UserDataQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::TimeBasedFilterQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@TimeBasedFilterQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::PresentationQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@PresentationQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::PartitionQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@PartitionQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::TopicDataQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@TopicDataQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::GroupDataQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@GroupDataQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::HistoryQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@HistoryQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::ResourceLimitsQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@ResourceLimitsQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::DurabilityServiceQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@DurabilityServiceQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::LifespanQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@LifespanQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::DataRepresentationQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@DataRepresentationQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::TypeConsistencyEnforcementQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@TypeConsistencyEnforcementQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::DisablePositiveACKsQosPolicy::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@DisablePositiveACKsQosPolicy@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::TypeIdV1::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@TypeIdV1@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
error LNK2001: unresolved external symbol "public: virtual bool __cdecl eprosima::fastrtps::TypeObjectV1::addToCDRMessage(struct eprosima::fastrtps::rtps::CDRMessage_t *)" (?addToCDRMessage@TypeObjectV1@fastrtps@eprosima@@UEAA_NPEAUCDRMessage_t@rtps@23@@Z)
```

I'm not verse with cmake, so I only used it to build the dependencies and fastrtps lib. Then made a new visual studio projects and manually linked the includes and libs accordingly.

--- a/include/fastrtps/qos/ParameterTypes.h
+++ b/include/fastrtps/qos/ParameterTypes.h
@@ -146,7 +146,7 @@ public:
      * @param[in,out] msg Pointer to the message where the parameter should be added.
      * @return True if the parameter was correctly added.
      */
-    virtual bool addToCDRMessage(rtps::CDRMessage_t* msg) = 0;
+    virtual bool RTPS_DllAPI addToCDRMessage(rtps::CDRMessage_t* msg) = 0;
 
 public:
     //!Parameter ID
@@ -223,7 +223,7 @@ class ParameterLocator_t: public Parameter_t
 /**
  *
  */
-class ParameterString_t: public Parameter_t 
+class ParameterString_t: public Parameter_t
 {
     public:
         ParameterString_t(){};

--- a/include/fastrtps/qos/ParameterTypes.h
+++ b/include/fastrtps/qos/ParameterTypes.h
@@ -146,7 +146,7 @@ public:
      * @param[in,out] msg Pointer to the message where the parameter should be added.
      * @return True if the parameter was correctly added.
      */
-    virtual bool RTPS_DllAPI addToCDRMessage(rtps::CDRMessage_t* msg) = 0;
+    RTPS_DllAPI virtual bool addToCDRMessage(rtps::CDRMessage_t* msg) = 0;
 
 public:
     //!Parameter ID

